### PR TITLE
#247 SaaS deployment documentation

### LINKED ISSUE #247: Fix the documentation following the introduction of SaaS features.
Due are two kinds of changes.
1. The language container deployment has been moved to the python-extension-common. So has the user guide in this part. This user guide needs to reference the other one.
2. There is patchy documentation of the new features related to the SaaS deployment. This needs to be sorted out.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #247

### All Submissions:

* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")
* [ ] Before you merge don't forget to run tests in AWS CodeBuild, by adding `[CodeBuild]` to the commit message

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -51,13 +51,14 @@ models on the Exasol Cluster. More information on The BucketFS can be found
...
-  - An Exasol connection object must be created with Exasol BucketFS connection 
-  information and credentials. 
-  - A format of the connection object is as following: 
+
+### Exasol DB
+- The Exasol cluster must already be running with version 7.1 or later.
+- DB connection information and credentials are needed.
+
+### BucketFS Connection 
+An Exasol connection object must be created with Exasol BucketFS connection  information and credentials.
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:60]:
```suggestion
An Exasol connection object must be created with Exasol BucketFS connection information and credentials. 
```

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -51,13 +51,14 @@ models on the Exasol Cluster. More information on The BucketFS can be found
...
-  information and credentials. 
-  - A format of the connection object is as following: 
+
+### Exasol DB
+- The Exasol cluster must already be running with version 7.1 or later.
+- DB connection information and credentials are needed.
+
+### BucketFS Connection 
+An Exasol connection object must be created with Exasol BucketFS connection  information and credentials. 
+A format of the connection object is as following:
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:61]:
```suggestion
The format of the connection object is as following: 
```

File: doc/user_guide/user_guide.md [hunk 3]
```diff
@@ -164,22 +166,35 @@ poetry build
...
 
   ```buildoutcfg
   python -m exasol_transformers_extension.deploy language-container <options>
   ```
 
+Please refer to the [Language Container Deployment Guide] for details about this command. 
+
+### Scripts Deployment
+
+Next you need to deploy all necessary scripts installed in the previous step to the specified
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:185]: > installed in the previous step

this part I don't understand

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -164,22 +166,35 @@ poetry build
...
   ```buildoutcfg
   python -m exasol_transformers_extension.deploy language-container <options>
   ```
 
+Please refer to the [Language Container Deployment Guide] for details about this command. 
+
+### Scripts Deployment
+
+Next you need to deploy all necessary scripts installed in the previous step to the specified 
+`SCHEMA` in your Exasol DB with the same `LANGUAGE_ALIAS`  using the following Python CLI command:
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:186]:
```suggestion
`SCHEMA` in your Exasol DB with the same `LANGUAGE_ALIAS` using the following Python CLI command:
```

File: doc/user_guide/user_guide.md [hunk 5]
```diff
@@ -396,20 +284,33 @@ You can invoke the python script as below which allows to load the transformer
...
+Unless stated otherwise in the comments column, the option is required for either or both backends.
+
+| Option name                  | On-Prem | SaaS | Comment                                         |
+|:-----------------------------|:-------:|:----:|:------------------------------------------------|
+| bucketfs-name                |   [x]   |      |                                                 |
+| bucketfs-host                |   [x]   |      |                                                 |
+| bucketfs-port                |   [x]   |      |                                                 |
+| bucketfs-user                |   [x]   |      |                                                 |
+| bucketfs-password            |   [x]   |      | Env. [BUCKETFS_PASSWORD]                        |
+| bucketfs-use-https           |   [x]   |      | Optional boolean, defaults to False             |
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:302]: this default we maybe should change

File: doc/user_guide/user_guide.md [hunk 6]
```diff
@@ -396,20 +284,33 @@ You can invoke the python script as below which allows to load the transformer
...
+| bucket                       |   [x]   |      |                                                 |
+| saas-url                     |         | [x]  |                                                 |
+| saas-account-id              |         | [x]  | Env. [SAAS_ACCOUNT_ID]                          |
+| saas-database-id             |         | [x]  | Optional, Env. [SAAS_DATABASE_ID]               |
+| saas-database-name           |         | [x]  | Optional, provide if the database_id is unknown |
+| saas-token                   |         | [x]  | Env. [SAAS_TOKEN]                               |
+| model-name                   |   [x]   | [x]  |                                                 |
+| path-in-bucket               |   [x]   | [x]  | Root location in the bucket for all models      |
+| sub-dir                      |   [x]   | [x]  | Sub-directory where this model should be stored |
+| [no_]use-ssl-cert-validation |   [x]   | [x]  | Optional boolean, defaults to True              |
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:312]: the default should be changed as well at some point

File: doc/user_guide/user_guide.md (hunk 3 above)
Comment (ahsimb) [reply doc/user_guide/user_guide.md:185]: Neither do I. Let's delete this bit.

**Comment by ahsimb** [comment]
On a general note, I am not sure that this "personalized" (not sure if it's a proper English term) form, when we say "**You** need ..., **You** should ..., etc" is appropriate in a user manual.

File: doc/user_guide/user_guide.md (hunk 5 above)
Comment (ahsimb) [reply doc/user_guide/user_guide.md:302]: Let's do this now.

File: doc/user_guide/user_guide.md (hunk 6 above)
Comment (ahsimb) [reply doc/user_guide/user_guide.md:312]: Why? We would normally (except when using the Docker-DB) want the validation to be on.

Comment (tkilias) [reply doc/user_guide/user_guide.md:312]: ah, I can't read, cert validation is per default on, perfect

**Comment by tkilias** [comment]
> On a general note, I don't know if this "personalized" (not sure if it's the right English term) form, when we say "**You** need ..., **You** should ..., etc" is appropriate in a user manual. It sounds a bit patronizing to me. But if we want to change this we should probably open another ticket.

Hm, good question. For non-native speakers, it is often difficult to decide. Maybe, we should have a look at what are the recommendation, for how to write documentation.

File: doc/user_guide/user_guide.md [hunk 7]
```diff
@@ -299,7 +299,7 @@ Unless stated otherwise in the comments column, the option is required for eithe
 | bucketfs-port                |   [x]   |      |                                                 |
 | bucketfs-user                |   [x]   |      |                                                 |
 | bucketfs-password            |   [x]   |      | Env. [BUCKETFS_PASSWORD]                        |
-| bucketfs-use-https           |   [x]   |      | Optional boolean, defaults to False             |
+| bucketfs-use-https           |   [x]   |      | Optional boolean, defaults to True              |
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:302]: this also needs to be changed in the code

Comment (ahsimb) [reply doc/user_guide/user_guide.md:302]: Yes, of course. I changed it in the other PR, which is currently in review.

Comment (tkilias) [reply doc/user_guide/user_guide.md:302]: Ah ok

**Review by tkilias — APPROVED** [review]

**Pull request merged**